### ISSUE TITLE
Fixed effects missing in MixMIL.predict

### ISSUE THREAD

**Issue by dominiksturm** [AIH-SGML/MixMIL#18]
Hey, 

looking at the example codes provided it seems like the `predict` function computes the logits of the model given some input. However, upon closer inspection we realized that the function does not include the fixed effects of the model, which can have an influence even if it's just an intercept and also does not reflect the logits of the model used during training comparing l.99 and l.177 in `model.py`. By manually including  the intercept like the following true positives improve significantly:
```python
u = model.predict(test_Xs, scaling=(mean,std))
f = test_F.mm(model.alpha.detach())

y_pred = torch.sigmoid(u + f).cpu().numpy()
```

This seems to be a bug in the current `predict` function or am I missing something?

Best,
Dominik

**Comment by jan-engelmann** [comment]
Hi Dominik,
thanks for looking into it!
This depends on the what kind of fixed effects you're using and which question you're asking.
If your fixed effects are in themselves predictive of your outcome: yes the prediction performance will be better

But if your fixed effects are bag covariates which you do not want to predict from (like in genetics population structure or gender), then you should not include them in the predict function.

But point taken. It'd be useful to give the user the option to use them, right?

Cheers,
Jan

**Comment by dominiksturm** [comment]
Hey Jan, thank you for the fast answer! I see, that makes sense. Giving the option would be nice, to avoid confusion. Happy holidays, Dominik
